include/swift/DependencyScan/DependencyScanningTool.h

```diff
@@ -24,26 +24,34 @@
 namespace swift {
 namespace dependencies {
 class DependencyScanningTool;
+class DependencyScanDiagnosticCollector;
+
+struct ScanQueryInstance {
+  std::unique_ptr<CompilerInstance> ScanInstance;
```

Comment:
Since two fields are always created together, should `std::unique_ptr` goes onto `ScanQueryInstance` when needed and make this:
```
struct {
CompilerInstance ScanInstance;
DependencyScanDiagnosticCollector Diags;
}
```

Reply:
The reason it's currently a struct of two pointers is because command-line scans will not set `ScanDiagnostics`, because they are okay with just using the parent instance's diagnostic reporting infrastructure. 

Furthermore, while it isn't configured to do so now, I'd like to follow-up with emitting diagnostics when we fail to create a `ScanInstance` for some reason (e.g. unknown argument?), so it would be nice to be able to have the instance fail to set, but still have collected diagnostics.